File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/doctors/referral-source/referral-source.model.ts
import {
  NgForm,
  FormGroup,
  FormControl,
  Validators,
  FormBuilder,
  ReactiveFormsModule
} from '@angular/forms';
import * as moment from 'moment/moment';
export class ReferralSource {
  public ReferralSourceId: number = 0;
  public PatientId: number = 0;
  public Newspaper: boolean;
  public Unknown: boolean;
  public Doctor: boolean;
  public Radio: boolean;
  public WebPage: boolean;
  public FriendAndFamily: boolean;
  public Staff: boolean;
  public Magazine: boolean;
  public TV: boolean;
  public Others: string = null;
  public CreatedDate: string = "";
  public Note: string = null;
  public CreatedBy: number = null;
  public ModifiedBy: number = null;
  public CreatedOn: string = null;
  public ModifiedOn: string = null;
  public ReferralSourceValidator: FormGroup = null;

  constructor() {
    var _formBuilder = new FormBuilder();
    this.ReferralSourceValidator = _formBuilder.group({
      //'Note': ['', Validators.compose([Validators.required,Validators.maxLength(200)])],
      //'Occupation': ['', Validators.compose([Validators.required, Validators.maxLength(50)])],
    });
  }

  public IsDirty(fieldName): boolean {
    if (fieldName == undefined)
      return this.ReferralSourceValidator.dirty;
    else
      return this.ReferralSourceValidator.controls[fieldName].dirty;
  }

  public IsValid(): boolean { if (this.ReferralSourceValidator.valid) { return true; } else { return false; } } public IsValidCheck(fieldName, validator): boolean {
    if (fieldName == undefined)
      return this.ReferralSourceValidator.valid;
    else
      return !(this.ReferralSourceValidator.hasError(validator, fieldName));
  }

}

File: Code/Components/DanpheEMR.ServerModel/ClinicalModels/ReferralSourceModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class ReferralSource
    {
        [Key]
        public int ReferralSourceId { get; set; }
        public int PatientId { get; set; }
        public bool Newspaper { get; set; }
        public bool Unknown { get; set; }
        public bool Doctor { get; set; }
        public bool Radio { get; set; }
        public bool WebPage { get; set; }
        public bool FriendAndFamily { get; set; }
        public bool Staff { get; set; }
        public bool TV { get; set; }
        public string Others { get; set; }
        public bool Magazine { get; set; }
        public string Note { get; set; }
        public int? CreatedBy { get; set; }
        public int? ModifiedBy { get; set; }
        public DateTime? CreatedOn { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public virtual PatientModel Patient { get; set; }
    }
}
